src/software/thunderscope/robot_diagnostics/robot_view.py

```diff
@@ -10,6 +12,32 @@
 from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
 
 
+class RobotCrashDialog(QDialog):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self.setWindowTitle("Robot Crash")
+
+        self.buttonBox = QDialogButtonBox(QDialogButtonBox.StandardButton.Ok)
+        self.buttonBox.accepted.connect(self.accept)
+        self.buttonBox.rejected.connect(self.reject)
+
+        self.layout = QVBoxLayout()
+        self.message = QLabel("Placeholder")
+        self.layout.addWidget(self.message)
+        self.robot_status = RobotStatusView()
+        self.layout.addWidget(self.robot_status)
+        self.layout.addWidget(self.buttonBox)
+        self.setLayout(self.layout)
+
+    def setMessage(self, message):
+        self.message.setText(message)
+
+    def setRobotStatus(self, robot_status):
```

Comment:
can you add some docs for these functions?

Reply:
added